### PR TITLE
[PLAT-693] Make embedded wiki images viewable through a view-only link

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Currently using a view-only link will break images or other files embedded in the wiki. This PR adds the view_only param into the embedded file links allowing the file to be viewed.

## Changes

- simple JS changes to the grab the image links and append the view_only param
- adds a new conditional to the markdown to add the view_only param to the embedded files.

## QA Notes

Create Wiki with various embedded files in a private project, then create a view-only link and try to view those files while logged out with the view-only link. You should be able to see them now.

Also remember to use the various sizing options available! 

## Documentation

🐞 fix, no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/PLAT-693

### CODE REVIEW

File: website/templates/project/project_base.mako [hunk 1]
```diff
@@ -214,6 +214,7 @@
             registrationMetaSchemas: ${ node['registered_schemas'] | sjson, n },
             registrationMetaData: ${ node['registered_meta'] | sjson, n },
             contributors: ${ node['contributors'] | sjson, n },
+            viewOnlyLink: ${ node['link'] | sjson, n }
```
Comment (sloria) [inline website/templates/project/project_base.mako]: Why does this need to be added to contextVars? Can you just access it from the context vars?

File: addons/wiki/static/wikiPage.js [hunk 2]
```diff
@@ -105,6 +105,12 @@ function ViewWidget(visible, version, viewText, rendered, contentURL, allowMathj
                             self.rendered(self.renderMarkdown(rawContent));
                             $markdownElement.css('display', 'inherit');
                         }
+                        if (window.contextVars.node.viewOnlyLink) {
+                            $($markdownElement).children('p').children('img').each(function(idx, elm){
+                                elm.setAttribute('src', elm.src + '&view_only=' + window.contextVars.node.viewOnlyLink);
```
Comment (sloria) [inline addons/wiki/static/wikiPage.js]: This doesn't seem quite right. This will add the `view_only` query to *every* img tag in the markdown content (not just OSF embeds).

Why not modify the the function used to build the MFR in `markdown.js`?

File: website/static/js/markdown.js [hunk 3]
```diff
@@ -31,11 +32,33 @@ var bootstrapTable = function(md) {
 var oldMarkdownList = function(md) {
     md.block.ruler.after('hr', 'pyMarkdownList', pymarkdownList);
 };
+
+
+var ViewOnlyImage = function(md) {
```
Comment (sloria) [inline website/static/js/markdown.js]: Drive-by: Use `lowerCamelCase` for consistency with the other plugins in this file.

File: website/static/js/markdown.js [hunk 4]
```diff
@@ -31,11 +32,33 @@ var bootstrapTable = function(md) {
...
+    var defaultRenderer = md.renderer.rules.image;
+    md.renderer.rules.image = function (tokens, idx, options, env, self) {
+        var token = tokens[idx];
+        var imageLink = token.attrs[token.attrIndex('src')][1];
+        var WaterbuterRegex = new RegExp(window.contextVars.waterbutlerURL + 'v1\/resources\/[a-zA-Z0-9]{1,}\/providers\/[a-z0-9]{1,}\/');
```
Comment (sloria) [inline website/static/js/markdown.js]: Drive-by: This regex should only need to be instantiated once. Also, it should probably be a constant (and therefore use `UPPER_CASE`).

File: website/static/js/markdown.js [hunk 5]
```diff
@@ -68,6 +92,7 @@ var markdown = new MarkdownIt('commonmark', {
 var markdownQuick = new MarkdownIt('commonmark', { linkify: true })
     .use(require('markdown-it-sanitizer'))
     .use(require('markdown-it-imsize'))
+    .use(viewOnlyImage)
```
Comment (sloria) [inline website/static/js/markdown.js]: Nit: I'm not sure this plugin is necessary for `markdownQuick`. We only use `markdownQuick` while a user is editing the wiki, in which case they will already have read access to the images. But maybe it's better to have close parity between markdownQuick and markdownFull 🤔  . No big deal: this is fine as is.

**Comment by pattisdr** [comment]
heading back to staging3 again for another pass!
